### DISCUSSION TITLE
How can i implement registration endpoint

### DISCUSSION THREAD

**Discussion by SelenaWenna** [soofstad/react-oauth2-pkce#153] [General]
How can i implement registration endpoint? Have it feature in the library?

**Comment by soofstad** [comment]
Anything related to registration is not part of the OAuth2 spec, and so not covered by this library.

However, depending on how you registration work flow is configured, you might be able to solve it by calling `logIn` with extra parameters, indicating that it is a login request.
`logIn: (state?: string, additionalParameters?: { [key: string]: string | boolean | number }) => void`

Registration has been discussed before here; https://github.com/soofstad/react-oauth2-pkce/discussions/113.

If this approach does not work out, I'd be happy to discuss alternatives
